compaction/compaction_manager.cc

```diff
@@ -827,8 +827,8 @@ void compaction_manager::do_stop() noexcept {
     }
 }
 
-inline bool compaction_manager::can_proceed(replica::table* t) const {
-    return (_state == state::enabled) && _compaction_state.contains(&t->as_table_state()) && !_compaction_state.at(&t->as_table_state()).compaction_disabled();
+inline bool compaction_manager::can_proceed(compaction::table_state* t) const {
```

Comment:
Perhaps we should take the opportunity to switch to reference, AFAIK this pointer is never null.

Reply:
It is private to compaction_manager, and we use the pointer internally anyhow so converting back and forth between a pointer and a reference won't buy us anything.

Reply:
Well we are converting back-and-forth anyway because we have references in many entry points. I think if we are making changes we should moving towards references.

Reply:
This series is long enough, I like the approach of just doing straightforward translations and not risking anything.